src/API/Prometheus.queries.ts

```diff
@@ -1,97 +1,102 @@
+import { PrometheusLabelsV2, PrometheusMetricsV2 } from '@config/prometheus';
 import { IntervalTimePropValue } from '@sk-types/Prometheus.interfaces';
 
 import { AvailableProtocols, Quantiles } from './REST.enum';
 
 export const queries = {
-  // http request queries
-  getRequestRateByMethodInInTimeRange(param: string, range: IntervalTimePropValue) {
-    return `sum by(method)(rate(http_requests_method_total{${param}}[${range}]))`;
-  },
-
-  // http response queries
-  getResponseCountsByPartialCodeInTimeRange(param: string, range: IntervalTimePropValue) {
-    return `sum by(partial_code)(label_replace(sum_over_time(http_requests_result_total{${param}}[${range}]),"partial_code", "$1", "code","(.*).{2}"))`;
-  },
+  // data transfer queries
+  getByteRateByDirectionInTimeRange(param: string, range: IntervalTimePropValue, areDataReceived = false) {
+    const label = areDataReceived ? PrometheusMetricsV2.ReceivedBytes : PrometheusMetricsV2.SentBytes;
 
-  getResponseRateByPartialCodeInTimeRange(param: string, range: IntervalTimePropValue) {
-    return `sum by(partial_code)(label_replace(rate((http_requests_result_total{${param}}[${range}])),"partial_code", "$1", "code","(.*).{2}"))`;
+    return `sum(rate(${label}{${param}}[${range}]))`;
   },
 
   // latency queries
   getPercentilesByLeInTimeRange(param: string, range: IntervalTimePropValue, quantile: Quantiles) {
-    return `histogram_quantile(${quantile},sum(rate(flow_latency_microseconds_bucket{${param}}[${range}]))by(le))`;
+    return `histogram_quantile(${quantile},sum(rate(${PrometheusMetricsV2.LatencyBuckets}{${param}}[${range}]))by(le))`;
   },
 
   getBucketCountsInTimeRange(param: string, range: IntervalTimePropValue) {
-    return `sum by(le)(floor(delta(flow_latency_microseconds_bucket{${param}}[${range}])))`;
+    return `sum by(le)(floor(delta(${PrometheusMetricsV2.LatencyBuckets}{${param}}[${range}])))`;
   },
 
-  // data transfer queries
-  getByteRateByDirectionInTimeRange(param: string, range: IntervalTimePropValue) {
-    return `sum by(direction)(rate(octets_total{${param}}[${range}]))`;
+  // calculate the open connections serie
+  getActiveFlowsInTimeRange(paramSource: string) {
+    //TODO: active flows are supposed to be in and out. In case we don't have processes as filters these values are used for both in and out (duplicated)
+    // this is a temporary solution
+    const hasProcessParam =
+      paramSource.includes(`${PrometheusLabelsV2.SourceProcess}`) ||
+      paramSource.includes(`${PrometheusLabelsV2.DestProcess}`);
+    const divider = hasProcessParam ? '' : '/2';
+
+    return `sum(${PrometheusMetricsV2.TcpOpenOnnections}{${paramSource}})${divider}`;
```

Comment:
A few things here:

I do not think that the divider is needed anymore - a I counting connections instead of flows in each direction.

For TcpOpen`O`nnections, instead of `GAUGE active_flows` and a `COUNTER total_flows` metrics I chose `COUNTER skupper_connections_opened_total` and `COUNTER skupper_connections_closed_total` counters. The difference here would be something like this: `sum(${PrometheusMetricsV2.TcpOpenedConnections}{${paramSource}} - ${PrometheusMetricsV2.TcpClosedConnections}{${paramSource}})`